### PR TITLE
Feature/support json output

### PR DESCRIPTION
**Pull request by rshaiju**
- Added Json support
- Default serialization is the existing one(list)

### CODE REVIEW

**Review by Araballi — COMMENTED** [review]
sample project has to be modified

**Review by Araballi — COMMENTED** [review]
Can you remove "Console.WriteLine("WIP");" from sample project

**Comment by rshaiju** [comment]
> sample project has to be modified

Not sure. Bcoz, this output formatting changes are confined to the tool. Library still returns the same object that it used to return before except for some annotations on the enum property which has no impact on the sample application

File: sample/Kmd.Logic.Gateway.Automation.Sample/Program.cs [hunk 1]
```diff
@@ -65,8 +65,6 @@ private static async Task Run(AppConfiguration configuration)
             {
                 Console.WriteLine(result.ToString());
             }
-
```
Comment (NakulSeerpu) [inline sample/Kmd.Logic.Gateway.Automation.Sample/Program.cs:68]: Extra line, infact we don't need that console statement

File: src/Kmd.Logic.Gateway.Automation.Tool/PublishCommandHandler.cs [hunk 2]
```diff
@@ -16,10 +16,7 @@ public async Task<int> Handle(PublishCommand cmd)
             var gatewayAutomation = new GatewayAutomation(httpClient, this.logicTokenProviderFactory, this.gatewayOptions);
             var results = await gatewayAutomation.PublishAsync(cmd.FolderPath).ConfigureAwait(false);
 
-            foreach (var result in results)
-            {
-                Console.WriteLine(result.ToString());
-            }
+            this.outputFormatter.PrintResults(results);
 
             return 0;
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation.Tool/PublishCommandHandler.cs:21]: this code has ben changed in your previous PR, it seem code is not merged

File: src/Kmd.Logic.Gateway.Automation/ItemValidationResultBase.cs [hunk 3]
```diff
@@ -2,6 +2,8 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using Newtonsoft.Json;
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/ItemValidationResultBase.cs]: are these namespace in use ?

File: src/Kmd.Logic.Gateway.Automation.Tool/OutputFormatters/GatewayAutomationResultsExtension.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            dynamic result = new JObject();
+            result.Results = new JArray();
+            if (gatewayAutomationResults != null)
+            {
+                foreach (var gatewayAutomationResultItem in gatewayAutomationResults)
+                {
+                    dynamic resultItemJson = new JObject();
+
+                    if (gatewayAutomationResultItem.EntityId.HasValue)
```
[the remark this answers is not in the document]
Comment (rshaiju) [reply src/Kmd.Logic.Gateway.Automation.Tool/OutputFormatters/GatewayAutomationResultsExtension.cs:19]: you want me to remove the curly braces you mean?

File: src/Kmd.Logic.Gateway.Automation/ItemValidationResultBase.cs (hunk 3 above)
Comment (rshaiju) [reply src/Kmd.Logic.Gateway.Automation/ItemValidationResultBase.cs]: It is already fixed. 

image

File: src/Kmd.Logic.Gateway.Automation.Tool/OutputFormatters/GatewayAutomationResultsExtension.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (rshaiju) [reply src/Kmd.Logic.Gateway.Automation.Tool/OutputFormatters/GatewayAutomationResultsExtension.cs:19]: image

**Review by Araballi — COMMENTED** [review]
LGTM

**Review by Araballi — COMMENTED** [review]
Update release note and read me document  for output type

**Review by mlatoszek — DISMISSED** [review]
Approved

**Review by Araballi — APPROVED** [review]
LGTM

**Pull request merged**